Crash when trying to create NAT gateway

/kind bug

**What steps did you take and what happened:**
Followed the quick start guide but for some reason CAPA is unable to create NAT Gateways. I am assuming the problem is specific to my AWS account.

I can't quite tell what the issue is because there is a nil pointer deref that prevents CAPA from logging the error message. I believe the problem is `*out.NatGateway.NatGatewayId` here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/services/ec2/natgateways.go#L178. 

```
I1013 18:21:45.157634       1 listener.go:40] controller-runtime/metrics "msg"="metrics server is starting to listen"  "addr"=":8080"
I1013 18:21:45.159218       1 controller.go:121] controller-runtime/controller "msg"="Starting EventSource"  "controller"="awsmachine" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"ami":{}},"status":{"ready":false}}}
I1013 18:21:45.159537       1 controller.go:121] controller-runtime/controller "msg"="Starting EventSource"  "controller"="awsmachine" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"metadata":{},"bootstrap":{},"infrastructureRef":{}},"status":{"bootstrapReady":false,"infrastructureReady":false}}}
I1013 18:21:45.159835       1 controller.go:121] controller-runtime/controller "msg"="Starting EventSource"  "controller"="awsmachine" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"networkSpec":{"vpc":{}}},"status":{"network":{"apiServerElb":{"attributes":{}}},"bastion":{"id":""},"ready":false}}}
I1013 18:21:45.160447       1 controller.go:121] controller-runtime/controller "msg"="Starting EventSource"  "controller"="awscluster" "source"={"Type":{"metadata":{"creationTimestamp":null},"spec":{"networkSpec":{"vpc":{}}},"status":{"network":{"apiServerElb":{"attributes":{}}},"bastion":{"id":""},"ready":false}}}
I1013 18:21:45.160574       1 main.go:166] setup "msg"="starting manager"  
I1013 18:21:45.161290       1 leaderelection.go:235] attempting to acquire leader lease  capa-system/controller-leader-election-capa...
I1013 18:21:45.161797       1 internal.go:250] controller-runtime/manager "msg"="starting metrics server"  "path"="/metrics"
I1013 18:22:01.794398       1 leaderelection.go:245] successfully acquired lease capa-system/controller-leader-election-capa
I1013 18:22:01.795753       1 controller.go:134] controller-runtime/controller "msg"="Starting Controller"  "controller"="awscluster"
I1013 18:22:01.798203       1 controller.go:134] controller-runtime/controller "msg"="Starting Controller"  "controller"="awsmachine"
I1013 18:22:01.896475       1 controller.go:154] controller-runtime/controller "msg"="Starting workers"  "controller"="awscluster" "worker count"=2
I1013 18:22:01.900904       1 controller.go:154] controller-runtime/controller "msg"="Starting workers"  "controller"="awsmachine" "worker count"=2
I1013 18:22:01.903526       1 awscluster_controller.go:130] controllers/AWSCluster "msg"="Reconciling AWSCluster" "awsCluster"="capi-quickstart-2" "cluster"="capi-quickstart-2" "namespace"="default" 
E1013 18:22:57.849511       1 runtime.go:73] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 287 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x183f900, 0x2dc9390)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/runtime/runtime.go:69 +0x7b
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/runtime/runtime.go:51 +0x82
panic(0x183f900, 0x2dc9390)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2.(*Service).createNatGateway(0xc000134018, 0xc0005542a0, 0x18, 0x18, 0x2e00a80, 0x0)
	/workspace/pkg/cloud/services/ec2/natgateways.go:178 +0x270
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2.(*Service).reconcileNatGateways(0xc000134018, 0x0, 0x0)
	/workspace/pkg/cloud/services/ec2/natgateways.go:78 +0x5e4
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2.(*Service).ReconcileNetwork(0xc000134018, 0xc000134020, 0x1)
	/workspace/pkg/cloud/services/ec2/network.go:39 +0x1e7
sigs.k8s.io/cluster-api-provider-aws/controllers.reconcileNormal(0xc0004262a0, 0x1b39590, 0xc0004262a0, 0xc000657d70, 0x1dfc3a0)
	/workspace/controllers/awscluster_controller.go:142 +0x147
sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSClusterReconciler).Reconcile(0xc00056de90, 0xc000429b69, 0x7, 0xc000554200, 0x11, 0x2de1f00, 0x0, 0x0, 0x0)
	/workspace/controllers/awscluster_controller.go:99 +0x653
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004ea5a0, 0x18c26a0, 0xc000552760, 0x18c2600)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:216 +0x146
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004ea5a0, 0xc0000c4700)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:192 +0xb5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0004ea5a0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:171 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0005bf710)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/wait/wait.go:152 +0x54
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005bf710, 0x3b9aca00, 0x0, 0x1, 0xc0003dcb40)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0005bf710, 0x3b9aca00, 0xc0003dcb40)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:157 +0x311
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x148e6a0]

goroutine 287 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/runtime/runtime.go:58 +0x105
panic(0x183f900, 0x2dc9390)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2.(*Service).createNatGateway(0xc000134018, 0xc0005542a0, 0x18, 0x18, 0x2e00a80, 0x0)
	/workspace/pkg/cloud/services/ec2/natgateways.go:178 +0x270
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2.(*Service).reconcileNatGateways(0xc000134018, 0x0, 0x0)
	/workspace/pkg/cloud/services/ec2/natgateways.go:78 +0x5e4
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2.(*Service).ReconcileNetwork(0xc000134018, 0xc000134020, 0x1)
	/workspace/pkg/cloud/services/ec2/network.go:39 +0x1e7
sigs.k8s.io/cluster-api-provider-aws/controllers.reconcileNormal(0xc0004262a0, 0x1b39590, 0xc0004262a0, 0xc000657d70, 0x1dfc3a0)
	/workspace/controllers/awscluster_controller.go:142 +0x147
sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSClusterReconciler).Reconcile(0xc00056de90, 0xc000429b69, 0x7, 0xc000554200, 0x11, 0x2de1f00, 0x0, 0x0, 0x0)
	/workspace/controllers/awscluster_controller.go:99 +0x653
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004ea5a0, 0x18c26a0, 0xc000552760, 0x18c2600)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:216 +0x146
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004ea5a0, 0xc0000c4700)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:192 +0xb5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0004ea5a0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:171 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0005bf710)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/wait/wait.go:152 +0x54
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005bf710, 0x3b9aca00, 0x0, 0x1, 0xc0003dcb40)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0005bf710, 0x3b9aca00, 0xc0003dcb40)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190817020851-f2f3a405f61d/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:157 +0x311
```


**Environment:**

- Cluster-api-provider-aws version: v0.4.2
- Kubernetes version: (use `kubectl version`): v1.15.3
- OS (e.g. from `/etc/os-release`):

--- a/pkg/cloud/services/ec2/natgateways.go
+++ b/pkg/cloud/services/ec2/natgateways.go
@@ -175,7 +175,7 @@ func (s *Service) createNatGateway(subnetID string) (*ec2.NatGateway, error) {
 		}
 		return true, nil
 	}, awserrors.InvalidSubnet); err != nil {
-		record.Warnf(s.scope.AWSCluster, "FailedCreateNATGateway", "Failed to create new NAT Gateway %q: %v", *out.NatGateway.NatGatewayId, err)
+		record.Warnf(s.scope.AWSCluster, "FailedCreateNATGateway", "Failed to create new NAT Gateway: %v", err)
 		return nil, errors.Wrapf(err, "failed to create NAT gateway for subnet ID %q", subnetID)
 	}
 	record.Eventf(s.scope.AWSCluster, "SuccessfulCreateNATGateway", "Created new NAT Gateway %q", *out.NatGateway.NatGatewayId)

